generator/transit_generator.cpp

```diff
@@ -1,16 +1,146 @@
 #include "generator/transit_generator.hpp"
 
+#include "generator/osm_id.hpp"
+
+#include "routing_common/transit_serdes.hpp"
+#include "routing_common/transit_types.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_name_utils.hpp"
+#include "coding/file_writer.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/checked_cast.hpp"
 #include "base/logging.hpp"
 #include "base/macros.hpp"
 
+#include "3party/jansson/myjansson.hpp"
+
+using namespace platform;
+using namespace std;
+
+namespace
+{
+using namespace routing;
+using namespace routing::transit;
+
+/// \returns file name without extension by a file path if a file name which have zero, one of several extensions.
+/// For example,
+/// GetFileName("Russia_Nizhny Novgorod Oblast.transit.json") returns "Russia_Nizhny Novgorod Oblast"
+/// GetFileName("Russia_Nizhny Novgorod Oblast.mwm") returns "Russia_Nizhny Novgorod Oblast"
+string GetFileName(string const & filePath)
+{
+  string name = filePath;
+  my::GetNameFromFullPath(name);
+
+  string nameWithExt;
+  do
+  {
+    nameWithExt = name;
+    my::GetNameWithoutExt(name);
+  }
+  while (nameWithExt != name);
+
+  return name;
+}
+
+/// \brief Reads from |root| (json) and serialize an array to |serializer|.
+template <class Item>
+void SerializeObject(my::Json const & root, string const & key, Serializer<FileWriter> & serializer)
+{
+  vector<Item> items;
+
+  DeserializerFromJson deserializer(root.get());
+  deserializer(items, key.c_str());
+
+  for (auto const & s : items)
+    s.Visit(serializer);
+}
+}  // namespace
+
 namespace routing
 {
 namespace transit
 {
-void BuildTransit(std::string const & mwmPath, std::string const & transitDir)
+// DeserializerFromJson ---------------------------------------------------------------------------
+void DeserializerFromJson::operator()(m2::PointD & p, char const * name)
 {
-  LOG(LINFO, ("mwm path:", mwmPath, ", directory with transit:", transitDir));
+  json_t * pointItem = my::GetJSONOptionalField(m_node, name);
+  if (pointItem == nullptr)
```

Comment:
Are you sure that this is not an error?

Reply:
You're right. It's an error. For the time being there's on optional field in section transit. Let's exchange it with `json_t * pointItem = my::GetJSONObligatoryField(m_node, name);`